README.md

```diff
@@ -55,29 +55,60 @@ The criteria for picking these event names included:
  - It doesn't overlap with any known existing events, so introducing these could be done in parallel to the existing API in each framework, optionally deprecating and eventually removing the previous API.
 
+#### Reporting Order
+
+The recommended **reporting order** for emitting the aformentioned events with their related data should be done in **source order**. Please note that execution order is different from reporting order, that's why we don't assume any specific execution order, but we assume whatever is higher up in the source file should be emitted first.
+
+For more background check [#62](https://github.com/js-reporters/js-reporters/issues/62).
+
 ### Event Data
 
-Based on the discussion in [#12](https://github.com/js-reporters/js-reporters/issues/12#issuecomment-120483356), there are two basic data structures: Suites and Tests. A test represents an atomic test/spec/`it()`. A suite contains tests and optionally other suites. This can represent both flat structures like used by QUnit as well as nested suites like used by Jasmine or Mocha. The data structures are defined as follows:
-
- - **Test**: A test holds basic information on a single test/spec. It has the following set of required attributes:
-   - `testName`: name of the test
-   - `suiteName`: name of the suite the test belongs to
-   - `status`: result of the test. Can `passed`, `failed` or `skipped`. A skipped test is disabled, i.e. it will not be executed.
-   - `runtime`: execution time in milliseconds
-   - `errors`: array containing all errors. Depending on the framework, this is a single exception or a list of failed assertions. Will be empty for statuses other than failed.
- - **Suite**: A suite is a collection of tests and potentially other suites.
-   - `name`: name of the suite
-   - `childSuites`: array with all direct subsuites
-   - `tests`: array containing all tests that directly belong to the suite (but not to a child suite)
-   - `runtime`: execution time of the whole suite in milliseconds (including child suites)
-   - `status`: summarized status of the suite
-      - `failed`, if at least one test failed
-      - `skipped`, if all tests in the suite are skipped (and there is at least one skipped test)
-      - `passed`, if there is at least one passed test and all other tests are skipped or if there are no tests in the suite.
+Based on the discussion in [#12](https://github.com/js-reporters/js-reporters/issues/12#issuecomment-120483356), there are two basic data structures: Suites and Tests. A test represents an atomic test/spec/`it()`. A suite contains tests and optionally other suites.
+
+For `testStart` and `testEnd`, the corresponding test object is passed to the reporter. The same applies to `suiteStart` and `suiteEnd` where the matching suite object is passed to the reporter. For `runStart` and `runEnd` a "global" suite object is passed to the reporter, which contains all top-level suites as child suites.
```

Comment:
Hmm, I found it confusing, because we have defined what the `global suite" is:

>  a "global" suite object is passed to the reporter, which contains all top-level suites as child suites.

The global suite is not a top-level suite.